app/react/Markdown/components/PublicForm.js

```diff
@@ -0,0 +1,120 @@
+          <li className="wide">
+            <Control.text id="title" className="form-control" model=".title" />
+          </li>
+        </ul>
+      </FormGroup>
+    );
+  }
+
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.fileFields = [];
+    this.validators = Object.assign({ captcha: { required: val => val && val.length } }, validator.generate(props.template.toJS()));
+  }
+
+  attachDispatch(dispatch) {
+    this.formDispatch = dispatch;
+  }
+
+  resetForm() {
+    this.formDispatch(actions.reset('publicform'));
+    this.fileFields.forEach((input) => { input.value = ''; });
+  }
+
+  handleSubmit(_values) {
+    const values = { ..._values };
+    const { submit } = this.props;
+    values.file = _values.file ? _values.file[0] : undefined;
+    values.template = this.props.template.get('_id');
+    submit(values).then(() => {
+      this.refreshCaptcha();
+      this.resetForm();
+    });
+  }
+
+  renderCaptcha() {
+    return (
+      <FormGroup key="captcha" model=".captcha">
+        <ul className="search__filter">
+          <li><label><Translate>Are you a robot?</Translate><span className="required">*</span></label></li>
+          <li className="wide">
+            <Captcha refresh={(refresh) => { this.refreshCaptcha = refresh; }} model=".captcha"/>
+          </li>
+        </ul>
+      </FormGroup>
+    );
+  }
+
+  renderFileField(id, options) {
+    const defaults = { getRef: (node) => { this.fileFields.push(node); }, className: 'form-control', model: `.${id}` };
+    const props = Object.assign(defaults, options);
+    return (
+      <div className="form-group">
+        <ul className="search__filter">
+          <li>
+            <label htmlFor={id}>
+              <Translate>{id === 'file' ? 'File' : 'Attachments'}</Translate>
+              <Control.file id={id} {...props} />
+            </label>
+          </li>
+        </ul>
+      </div>
+    );
+  }
+
+  render() {
+    const { template, thesauris, file, attachments } = this.props;
+    return (
+      <LocalForm validators={this.validators} model="publicform" getDispatch={dispatch => this.attachDispatch(dispatch)} onSubmit={this.handleSubmit}>
+        {PublicForm.renderTitle()}
+        <MetadataFormFields thesauris={thesauris} model="publicform" template={template} />
+        {file ? this.renderFileField('file', { accept: '.pdf' }) : false}
+        {attachments ? this.renderFileField('attachments', { multiple: 'multiple' }) : false}
+        {this.renderCaptcha()}
+        <input type="submit" className="btn btn-success" value="Submit"/>
+      </LocalForm>
+    );
+  }
+}
+
+PublicForm.propTypes = {
+  file: PropTypes.bool.isRequired,
+  attachments: PropTypes.bool.isRequired,
+  template: PropTypes.instanceOf(Immutable.Map).isRequired,
+  thesauris: PropTypes.instanceOf(Immutable.List).isRequired,
+  submit: PropTypes.func.isRequired,
+};
+
+export const mapStateToProps = (state, props) => ({
```

Comment:
I think there is enough 'logic' in this to merit a test.  Also, since these functions are usually exported only for testing purposes, I was expecting to find such tests.